src/server/Adaptive.ReactiveTrader.Server.IntegrationTests/ExecuteTradeIntegrationTests.cs

```diff
@@ -33,30 +29,18 @@ public ExecuteTradeIntegrationTests()
                                        .Publish()
                                        .RefCount();
 
-            _executionServiceInstance = _heartbeatStream
-                .Where(hb => hb.Type == ServiceTypes.Execution)
-                .Select(hb => hb.Instance)
-                .Take(1)
-                .Timeout(ResponseTimeout)
-                .ToTask()
-                .Result;
-
-            _timeoutCancellationTokenSource = new CancellationTokenSource();
+            _executionServiceInstance = GetInstanceName(ServiceTypes.Execution).Result;
```

Comment:
I know it used Task.Result before, but it would be nice if we could make this async.
I wonder if we can do it outside of the constructor maybe?

Reply:
I think that this test could benefit from bigger refactoring, but didn't want to get carried away